contributed_definitions/nyaml/SPM/NXspm.yaml

```diff
@@ -0,0 +1,832 @@
+              specific enough to resolve the version.
+      real_time_controller(NXrcs):
+        exists: optional
+        doc: |
+          The real-time controller information.
+        rcs_frequency(NX_NUMBER):
+          exists: optional
+          doc: |
+            The frequency of the real-time controller system which indicated the number of close-loop process
+            (gathering data, process data and update system) control cycles per second.
+      lockin_amplifier(NXlockin):
+        exists: optional
+        doc: |
+          The lock-in amplifier information.
+      scan_environment(NXenvironment):
+        doc: |
+          Information of the scan environment.
+        tip_temp(NX_NUMBER):
+          exists: optional
+          unit: NX_TEMPERATURE
+          doc: |
+            Temperature of STM head. Note: At least one field from stm_head_temp,
+            cryo_bottom_temp and cryo_sheild_temp must be provided.
+        cryo_bottom_temp(NX_NUMBER):
+          exists: optional
+          unit: NX_TEMPERATURE
+          doc: |
+            Temperature of liquid helium cryostat. Note: At least one field from
+            stm_head_temperature, cryo_bottom_temp and cryo_sheild_temp must be provided.
+        cryo_shield_temp(NX_NUMBER):
+          exists: optional
+          unit: NX_TEMPERATURE
+          doc: |
+            Temperature of liquid nitrogen shield. Note: At
+            least one field from stm_head_temperature, cryo_bottom_temp and cryo_sheild_temp.
+        scan_name:
+          exists: optional
+          doc: |
+            The name of the scan, mainly lab defined, that helps to differentiate among different
+            scans performed in lab.
+        current_sensor(NXsensor):
+          exists: optional
+          doc: |
+            This is a link to the current_sensor under the instrument group:
+            entry/experiment_instrument/current_sensor.
+        volatage_sensor(NXsensor):
+          exists: optional
+          doc: |
+            This is a link to the voltage_sensor under the instrument group:
+            entry/experiment_instrument/voltage_sensor.
+        piazo_sensor(NXsensor):
+          exists: optional
+          doc: |
+            This is a link to the piazo_sensor under the instrument group:
+            entry/experiment_instrument/piazo_sensor.
+        (NXscan_control):
+          doc: |
+            The scan control information like scan region or phase space, type of scan (e.g.
+            mesh, spiral, etc.), and scan speed, etc. This group mainly stores the raw scan
+            data. For processed data or final experimental data would go to NXdata group.
+          scan_name:
+            exists: optional
+            doc: |
+              The name of the scan, mainly lab defined, that helps to differentiate among different
+              scans performed in lab. This field is intended for several types of scan control
+              run under the same environment.
```

Comment:
there is already scan_name defined under scan_environment, do we need a copy here? Or maybe just a reference?